datagateway_api/src/search_api/query_filter_factory.py

```diff
@@ -175,16 +172,14 @@ def get_where_filter(where_filter_input, entity_name):
                 )
         else:
             log.info("Basic where filter found, extracting field, value and operation")
-            filter_data = SearchAPIQueryFilterFactory.get_condition_values(
+            field, value, operation = SearchAPIQueryFilterFactory.get_condition_values(
                 where_filter_input,
             )
-            where_filters.append(
-                SearchAPIWhereFilter(
-                    field=filter_data[0],
-                    value=filter_data[1],
-                    operation=filter_data[2],
-                ),
-            )
+
+            # Ignore filters on `isPublic`` fields as data is always public. This is
+            # hardcoded in the `models.py` module.
+            if field != "isPublic":
+                where_filters.append(SearchAPIWhereFilter(field, value, operation))
```

Comment:
I've just had a play around with JPQL queries and you might be able to have a skip of 1 then a limit of 0 for this to work:

`SELECT o FROM Investigation o INCLUDE o.keywords, o.type LIMIT 1, 0` - this returns an empty list

Reply:
This does the job but I discovered that it errors on the count endpoints because of the skip filter which I guess ICAT doesn't like and therefore returns and empty list (`[]`), causing an `IndexError` in https://github.com/ral-facilities/datagateway-api/blob/d8725abe9086dea1b9d1867ff2d2f37f52f32ddc/datagateway_api/src/search_api/helpers.py#L127
I tested this on the DataGateway API's `/investigations` endpoint and I get the same error - `http://{{datagateway-api}}/datagateway-api/investigations/count?skip=1`.

Reply:
This issue only occurs when filtering for `isPublic` with `False` values on count endpoints but works fine if filtering for `True` values. Given that the Search API only returns public data, the result for anyone filtering for `False` values will always be 0 so I don't see such filter being used. As a result of this, I believe that it is best to merge the code as it is.